### ISSUE TITLE
[Tutorial] Build and install openzfs 2.2.5 and Linux 6.9.10

### ISSUE THREAD

**Issue by soi1uhtgnei2urtonwe4u** [archzfs/archzfs#548]
1. Remove zfs (or your boot menu may be empty and you'd need to `arch-chroot` to fix it if you'd run the 2. command without removing zfs first):
`sudo pacman -R zfs-dkms zfs-utils`

2. `sudo downgrade linux linux-headers` to 6.9.10. When `downgrade` then asks to add the 2 packages to `pacman`'s `IgnorePkg`, confirm with [Y]es, to avoid accidentally installing newer Linux version on your next `# pacman -Syu` system update.
For NV users: Switching from `nvidia` to `nvidia-dkms` works for me, or my boot would stop at `Starting LightDM ..` and the `Xorg.0.log` would say

```
[     9.393] (EE) NVIDIA: Failed to initialize the NVIDIA kernel module. Please see the
[     9.393] (EE) NVIDIA:     system's kernel log for additional error messages and
[     9.393] (EE) NVIDIA:     consult the NVIDIA README for details.
```
: `sudo pacman -S nvidia-dkms` (it will auto detect a conflict and ask if you want to remove the `nvidia` package, confirm with [Y]es) 

3. Import gpg keys listed in the `zfs-dkms`/`PKGBUILD` file:
```
gpg --receive-keys 4F3BA9AB6D1F8D683DC2DFB56AD860EED4598027
gpg --receive-keys C33DF142657ED1F7C328A2960AB9E991C6AF658B
```

4. Build https://aur.archlinux.org/packages/zfs-utils:
[pc zfs-utils] `makepkg`

5. Build https://aur.archlinux.org/packages/zfs-dkms:
[pc zfs-dkms] `makepkg`

6. Installing `zfs-utils` and `zfs-dkms`:
`sudo pacman -U zfs-utils-2.2.5-1-x86_64.pkg.tar.zst zfs-dkms-2.2.5-1-any.pkg.tar.zst`
(`zfs-utils` needs to be installed first)

7. Done, no need to restart. This works for me:
[pc ~] `sudo zpool import <pool name> `
(before I had to use `sudo zpool export <my pool name> && sudo zpool import <my pool name>`)

**Comment by soi1uhtgnei2urtonwe4u** [comment]
OpenZFS v2.2.5 officially only supports up to Linux 6.9, I don't recommend going beyond this version.

**Comment by n0xena** [comment]
1) downgrade is an AUR-only package - anything "halfway official" should not rely on AUR packages - if you don't know how to manually switch to the desired kernel you better should not use zfs in such a way
2) why DKMS? for both zfs and nvidia - dkms itself ist yet another framework for kernel modules and just automates the rebuild on kernel update - if you focus on a specific kernel version just build the module for that version - no need for dkms at all
3) no reboot required - yes, in difference to windows linux supports loadung (and unloading) of kernel modules - but zfs is more than just a kernel module: zed (zfs event daemon), zfs auto import and mount and boot, auto scrub - a better (and more consistent) way to me is after installing zfs configuring its systemd units and then do a full system reboot
4) aside from 2 - that whole nvidia stuff doesn't belong here, at all - each module should be dealt with on its own - this has the risk for nvidia related questions that doesn't belong here - if you experienced issues with conflicting nvidia and zfs that's more like an issue on your end and likely nothin with zfs nor nvidia

overall a rather bad "tutorial" as in "yea, works, but can be done better"

**Comment by soi1uhtgnei2urtonwe4u** [comment]
The `zfs-dkms` are `zfs-utils` also AUR, so adding `downgrade` is fine (I use EndeavourOS, so it was already there and I don't care). But this is the only thing I agree with u: The fewer AUR packages, the better.

The NV stuff definitely belongs here, because every NV user will get the issue. And my tutorial is not about some idealistic, arch-only, unrealistic stuff. 

Try without dkms.

**Comment by n0xena** [comment]
For the sake of argue:
Each module should be dealt with on it's own!
Currently there no known conflicts between zfs and nvidia - so, no, it surely does NOT belong here. If, at all, it belongs upstream to the zfs repo if there is any conflict between the two - as this very repo here is just a couple of shell scripts to take the upstream release and build arch packages from it.
Also for the sake of testing:
As I do have a system with an nvidia gpu I tested on bare metal and encountered any issues or conflicts with the non-dkms modules. Along with that as you use Endeavour instead of vanilla Arch (which should be used for a tutorial, only - otherwise please update both the title and your initial post to reflect that!) this hints it could be a possible issue on your end - as it works fine with vanilla on bare metal.

Overall: Ok, I do agree with the courage to provide assistance, so do I - but as we're dealing with a kernel module here it should focus on that. You stuff here is too specific for your setup instead of generic which can lead others using vanilla or even another fork of arch and different setups failing for specifics not apply to them.
Consider a re-write with bare metal test with vanilla arch and with nvidia stuff to generalize your "tutorial" towards a more generic help.

**Comment by jzvikart** [comment]
@n0xena You need to understand that the entire purpose of this repo is for people to help other people, not to prove that they are right. If you're not contributing anything you might as well keep your opinions to yourself.

**Comment by r-maerz** [comment]
For anyone finding this, please note that I have created an archiso that includes both the LTS kernel as well as the openzfs modules for it. In case you want to avoid unsafe methods like the above, you can just boot that archiso instead.

It's available on [GitHub](https://github.com/r-maerz/archlinux-lts-zfs) for anyone to use.

**Comment by n0xena** [comment]
request for this to be closed as neither 2.2.5 nor 6.9 are relevant anymore
